cpu/native/native_cpu.c

```diff
@@ -225,7 +225,7 @@ void thread_yield_higher(void)
         irq_enable();
     }
     else {
-        isr_thread_yield();
+        sched_context_switch_request = 1;
```

Comment:
It's not correct, see https://github.com/RIOT-OS/RIOT/pull/6660/files#diff-ae383327c6bd21463d3010729a9284b3R221